api/src/main/java/org/apache/iceberg/PartitionField.java

```diff
@@ -19,26 +19,49 @@
 package org.apache.iceberg;
 
 import java.io.Serializable;
+import java.util.List;
+import java.util.stream.Collectors;
 import org.apache.iceberg.relocated.com.google.common.base.Objects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.transforms.Transform;
 
 /** Represents a single field in a {@link PartitionSpec}. */
 public class PartitionField implements Serializable {
-  private final int sourceId;
+  private final List<Integer> sourceIds;
   private final int fieldId;
   private final String name;
   private final Transform<?, ?> transform;
 
+  /**
+   * @deprecated since 1.10.0, will be removed in 2.0.0; use PartitionField(sourceIds, fieldId,
+   *     name, transform) instead.
+   */
+  @Deprecated
```

Comment:
I don't actually have a problem with keeping this method? It's essentially just going to save callers doing a List.of(element) and I don't think we are are really adding much complexity?

Reply:
That's a good point. The reason why I flagged deprecated is to "force" the callers to use source-ids "style" (I don't deprecated in the builder as the current transforms support only one argument).
I'm OK to remove the `@Deprecated` here.

Reply:
I removed the deprecation.